tests/phpunit/tests/http/http.php

```diff
@@ -661,4 +661,56 @@ function ( $response, $parsed_args, $url ) use ( &$pre_http_request_filter_has_r
 		$this->assertSame( 'PASS', wp_remote_retrieve_body( $redirect_response ), 'Redirect response body is expected to be PASS.' );
 		$this->assertTrue( $pre_http_request_filter_has_run, 'The pre_http_request filter is expected to run.' );
 	}
+
+	/**
+	 * Test that WP_Http::normalize_cookies method correctly casts integer keys to string.
+	 * @ticket 58566
+	 *
+	 * @covers WP_Http::normalize_cookies
+	 */
+	public function test_normalize_cookies_casts_integer_keys_to_string() {
+		$http = _wp_http_get_object();
+
+		$cookies = array(
+			'1'   => 'foo',
+			2     => 'bar',
+			'qux' => 7,
+		);
+
+		$cookie_jar = $http->normalize_cookies( $cookies );
+
+		$this->assertInstanceOf( 'WpOrg\Requests\Cookie\Jar', $cookie_jar );
+
+		foreach ( array_keys( $cookies ) as $cookie ) {
+			if ( is_string( $cookie ) ) {
```

Comment:
What's the purpose of checking for the string? `normalize_cookies` should now be handling it without either checking if it is a string or casting for it, or am I missing something?

Reply:
Since we are iterating over `$cookies` which still holds different type keys we need to cast to differentiate strings from non-strings, otherwise `$cookie_jar[ $cookie ]` below would not work for non-strings.

Reply:
nod nod, yeah, makes sense. I misread and thought it was iterating over the `$cookie_jar`. Disregard. Thanks!